misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/util/ConverterUtils.java

```diff
@@ -74,6 +84,77 @@ public static String escapeIdentifier(String identifier) {
         }
     }
 
+    /**
+     * This method returns existing Types on a module/file(for single file projects).
+     *
+     * @param filePath FilePath of the/a file in a singleFileProject or module
+     * @return {@link List<String>} List of already existing Types
+     */
+    public static List<String> getExistingTypeNames(String filePath) {
+        List<String> existingTypeNames = new ArrayList<>();
+        if (filePath == null) {
+            return existingTypeNames;
+        }
+
+        Path filePathResolved;
+        try {
+            filePathResolved = Path.of(filePath);
+        } catch (InvalidPathException e) {
+            return existingTypeNames;
+        }
+
+        Project project;
+        // Check if the provided file is a SingleFileProject
+        try {
+            project = SingleFileProject.load(filePathResolved);
+            List<Symbol> moduleSymbols =
+                    project.currentPackage().getDefaultModule().getCompilation().getSemanticModel().moduleSymbols();
+            moduleSymbols.forEach(symbol -> {
+                if (symbol.getName().isPresent()) {
+                    existingTypeNames.add(symbol.getName().get());
+                }
+            });
+        } catch (ProjectException pe) {
+            // Check if the provided file is a part of BuildProject
+            Path projectRoot = ProjectUtils.findProjectRoot(filePathResolved);
+            if (projectRoot != null) {
+                try {
+                    project = BuildProject.load(projectRoot);
+                    List<Symbol> moduleSymbols = project.currentPackage()
+                            .module(project.documentId(filePathResolved).moduleId())
+                            .getCompilation().getSemanticModel().moduleSymbols();
+                    moduleSymbols.forEach(symbol -> {
+                        if (symbol.getName().isPresent()) {
+                            existingTypeNames.add(symbol.getName().get());
+                        }
+                    });
```

Comment:
Shall we improve the logic to access the semantic model directly from the LS workspace manager, to avoid redundant project loadings in the tool side?